pkg/src/src/main/scala/edu/berkeley/cs/amplab/sparkr/SerDe.scala

```diff
@@ -282,3 +282,9 @@ object SerDe {
     value.foreach(v => writeBytes(out, v))
   }
 }
+
+object SerializationFormats {
```

Comment:
The nice things about case objects are that they automagically get a bunch of sane methods for free, including toString:
```
scala> case object ByteSerialized
defined module ByteSerialized

scala> ByteSerialized.toString
res0: String = ByteSerialized
```
So wherever we are doing `something == "row"` now, we could instead be more type safe by doing `something == RowSerialized`, and we don't have to worry about writing out the string literals anymore.

Besides this substitution, we just need to change the strings on the R side to match these new names. Thoughts on whether this sounds worthwhile?